src/main/cpp/JavaClassUtils.cpp

```diff
@@ -53,7 +53,7 @@ class JavaClassLoader : public JavaClass {
 
   jclass loadClass(JNIEnv *env, const char *class_name) {
     std::string binaryName = class_name;
-    for (int i = 0; i < binaryName.length(); ++i) {
+    for (unsigned i = 0; i < binaryName.length(); ++i) {
```

Comment:
I think that `size_t` would be better than `unsigned` here, since that is the exact type returned by `std::string::length()`.